utilities/src/bytes.rs

```diff
@@ -166,6 +199,22 @@ impl<'de, T: FromBytes> FromBytesDeserializer<T> {
     }
 }
 
+pub struct FromBytesUncheckedDeserializer<T: FromBytesUnchecked + FromBytes>(PhantomData<T>);
+
+impl<'de, T: FromBytesUnchecked + FromBytes> FromBytesUncheckedDeserializer<T> {
+    /// Deserializes a dynamically-sized byte array.
+    pub fn deserialize_with_size_encoding<D: Deserializer<'de>>(deserializer: D, name: &str) -> Result<T, D::Error> {
+        let mut buffer = Vec::with_capacity(32);
+        deserializer.deserialize_bytes(FromBytesVisitor::new(&mut buffer, name))?;
+
+        if UNCHECKED_DESERIALIZE.get() {
```

Comment:
Given the constraints at hand, seems like a good approach.

Does this mean that we don't need the `FromBytesUncheckedDeserializer`? Or is that still useful to ensure we don't have to implement `FromBytesUnchecked` on every object within snarkVM? Though that shouldn't be too hard, as they can always just call into `FromBytes`.

Reply:
Done in [a1e2f4d](https://github.com/ProvableHQ/snarkVM/pull/2789/commits/a1e2f4d151c2cc0705de5f97924c63bf76e90b2c).